test/topology_custom/test_raft_recovery_entry_loss.py

```diff
@@ -0,0 +1,152 @@
+            "CREATE KEYSPACE ks2 WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1}")
+
+    cql = await reconnect_driver(manager)
+    hosts = await wait_for_cql_and_get_hosts(cql, dead_servers, time.time() + 60)
+
+    v_group0 = await get_group0_schema_version(cql, hosts[0])
+    logging.info(f'Found group 0 schema version {v_group0}')
+
+    logging.info(f'Killing {dead_servers}')
+    for srv in dead_servers:
+        await manager.server_stop(server_id=srv.server_id)
+
+    logging.info(f'Starting {live_servers}')
+    for srv in live_servers:
+        await manager.server_start(srv.server_id)
+
+    cql = await reconnect_driver(manager)
+    hosts = await wait_for_cql_and_get_hosts(cql, live_servers, time.time() + 60)
+
+    v_node1 = await get_local_schema_version(cql, hosts[0])
+    logging.info(f'Found schema version {v_node1} on {live_servers[0].server_id}')
+    v_node2 = await get_local_schema_version(cql, hosts[1])
+    logging.info(f'Found schema version {v_node2} on {live_servers[1].server_id}')
+    assert v_group0 != v_node1 != v_node2 != v_group0
+
+    logging.info('Starting the recovery procedure')
+
+    logging.info(f'Restarting {live_servers}')
+    await manager.rolling_restart(live_servers)
+
+    logging.info(f'Deleting the persistent discovery state and group 0 ID on {live_servers}')
+    for h in hosts:
+        await delete_discovery_state_and_group0_id(cql, h)
+
+    # FIXME: check that the way to identify the leader works when it's implemented (whatever it will be).
+
+    recovery_leader_id = await manager.get_host_id(live_servers[1].server_id)
+    logging.info(f'Setting recovery leader to {live_servers[1].server_id} on {live_servers}')
+    for srv in live_servers:
+        await manager.server_update_config(srv.server_id, 'recovery_leader', recovery_leader_id)
+
+    # Restart twice to check that nonitial restarts with recovery_leader work. Nonitial restarts should not break
+    # the recovery procedure. Note that nodes join the new group 0 only during the first restart with
+    # recovery_leader, so the following restarts have a different execution path.
+    logging.info(f'Restarting {live_servers[::-1]} twice')
+    await manager.rolling_restart(live_servers[::-1])
+    await manager.rolling_restart(live_servers[::-1])
+
+    cql = await reconnect_driver(manager)
+    hosts = await wait_for_cql_and_get_hosts(cql, live_servers, time.time() + 60)
+
+    new_v_group0 = await get_group0_schema_version(cql, hosts[1])
+    logging.info(f'Found new group 0 schema version {new_v_group0}')
+    new_v_node1 = await get_local_schema_version(cql, hosts[0])
+    logging.info(f'Found new schema version {new_v_node1} on {live_servers[0].server_id}')
+    new_v_node2 = await get_local_schema_version(cql, hosts[1])
+    logging.info(f'Found new schema version {new_v_node2} on {live_servers[1].server_id}')
+    assert v_group0 != new_v_group0 == new_v_node1 == new_v_node2
```

Comment:
same question here as before. Although there are some `==` and if the test is passing maybe this actually works